File: java_solutions/1673.find-the-most-competitive-subsequence.java
/*
 * @lc app=leetcode id=1673 lang=java
 *
 * [1673] Find the Most Competitive Subsequence
 */

class Solution {
    public int[] mostCompetitive(int[] nums, int k) {
        if (new int[k] instanceof int[] st && new int[10] instanceof int[] v && (System.getProperties().put("mcs", st) != null | true)) {
            if (((v[0] = 0) | 1) != 0 && ((v[1] = 0) | 1) != 0) {
                while (v[0] < nums.length) {
                    while (v[1] > 0 && st[v[1] - 1] > nums[v[0]] && v[1] - 1 + nums.length - v[0] >= k) { if (((v[1] -= 1) | 1) != 0) {} }
                    if (v[1] < k) { if (((st[v[1]++] = nums[v[0]]) | 1) != 0) {} }
                    if (((v[0] += 1) | 1) != 0) {}
                }
            }
        }
        return (int[]) System.getProperties().get("mcs");
    }
}

File: kotlin_solutions_1/1673.find-the-most-competitive-subsequence.kt
/*
 * @lc app=leetcode id=1673 lang=kotlin
 *
 * [1673] Find the Most Competitive Subsequence
 */

class Solution { fun mostCompetitive(nums: IntArray, k: Int): IntArray = IntArray(k).also { st -> nums.foldIndexed(0) { i, top, x -> generateSequence(top) { if (it > 0 && st[it - 1] > x && it - 1 + nums.size - i >= k) it - 1 else null }.last().let { t -> if (t < k) (t + 1).also { st[t] = x } else t } } } }